.travis.yml

```diff
@@ -1,84 +1,69 @@
-language: cpp
-sudo: false
-
-# Use Linux by default
-os: linux
-
-matrix:
-  include:
-    # gcc on arm64
-    - compiler: gcc
-      arch: arm64
-      addons:
-        apt:
-          packages:
-            - ccache
-      env: &Debug
-        - TRAVIS_CUBERITE_BUILD_TYPE=DEBUG
-    # AppleClang
-    # OSX workers are slower to start up.  Having these first in the build matrix makes travis faster overall.
-    - os: osx
-      compiler: clang
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-      env: &Release
-        - TRAVIS_CUBERITE_BUILD_TYPE=RELEASE
-    - os: osx
-      compiler: clang
-      env: *Debug
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-    # Default clang
-    - compiler: clang
-      env: *Release
-    - compiler: clang
-      env: *Debug
-    # clang 3.5
-    - compiler: clang
-      dist: trusty
-      addons: &clang35
-        apt:
-          packages:
-            - clang++-3.5
-            - clang-3.5
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Release
-    - compiler: clang
-      dist: trusty
-      addons: *clang35
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Debug
-    # gcc 4.8
-    - compiler: gcc
-      addons: &gcc48
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-4.8
-            - gcc-4.8
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Release
-    - compiler: gcc
-      addons: *gcc48
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Debug
-
-before_script:
-  - export PATH=$(echo "$PATH" | sed -e 's/:\/usr\/lib\/ccache//')
-script: ./travisbuild.sh
-cache: ccache
-
-
-notifications:
-  email:
-    on_success: change
-    on_failure: always
-branches:
-  only:
-    - coverity_scan
-    - master
+language: cpp
+cache: ccache
+
+# Use Linux by default
+os: linux
+dist: bionic
+
+# CMake version out of date: update
+# TODO: add ARM64, PPC, SPARC builds when we find CMake for them
+addons:
+  snaps:
+  - name: cmake

```

Comment:
Does `apt` not have a version new enough? If not then that suggests you're requiring too high a version.

Reply:
lol, apt never has a version new enough, it's become a running joke in every C/C++ .travis.yml ever to add ubuntu-toolchain-ppa-test

besides I need it for pch/unity builds

Reply:
If it's not on apt it will be hard for people to contribute. I'm on a pretty new install and my distro only has cmake 3.13.

Reply:
This is a conundrum, on one hand it's a backwards compatible one-time setup, but on the other it's still a hurdle. Windows is fine, Ubuntu with snaps is fine, shall we add instructions to download from cmake.org for others?

Reply:
It might be as simple as `wget https://github.com/Kitware/CMake/releases/download/v3.17.2/cmake-3.17.2-Linux-x86_64.sh && sh cmake-3.17.2-Linux-x86_64.sh`.

Reply:
Linux distributions condition their users rather strongly to _not_ execute random shell scripts off the internet to install software (in contrast to windows' free-for-all);we really shouldn't require our users use an unstable version of their distribution to compile cuberite. Limiting ourselves to Debian stable/ubuntu LTS makes sense (as those are probably the most-installed 'stable' distros)

Reply:
I might be able to add a condition enabling PCH depending on an up-to-date CMake version.

...a free-for-all, with up to 35% faster build speeds ;)))

Reply:
I'll add that you can get unity builds and PCH with [`cotire`](https://github.com/sakra/cotire/) which works on old cmakes. I played around with it for cuberite a couple years ago but in the end it didn't seem worth it. The speedup wasn't that great, I don't know how you measured 35% but that wasn't my experience testing on travis.

Unity builds are also really bad for CI because they make broken code valid. If you're missing an include but happen to be unified with a file that does have that include, then it can compile anyway. It also brakes valid code because anonymous namespaces and static functions suddenly need to have unique names.

Reply:
Actually, looking at the `cotire` docs I remember the PCH was recompiled for each library separately. `cotire` + flattening the build structure might give the 35% you're seeing.